### PR TITLE
Make GGIR run easier on machines not configured for UK/US

### LINKED ISSUE #1047: Part 2 report stores some day names in local languages
I have always worked on English configured computer, but one GGIR user reported to me that this happens for them on their French computer always for the first day in the recording.

This is problematic because later parts will then struggle to recognise the day names.

Solution requires: Force language to English when defined day names in part 2.

### LINKED ISSUE #1048: sep_report and sep_config not sufficient for working with non-UK/US configured machines?
Arguments `sep_report` and `sep_config` were implemented a while ago by @jhmigueles to allow for saving csv reports with alternative column separator values to ease use in non UK/US configured R installations.

I tried it last week on a Dutch configured computer. Although columns seem to be split correctly the values were off, which makes me wonder whether it is also critical that we specify the `dec` argument every time the csv file is safe or loaded, e.g. in functions `extract_params` and `g.plot5`.

### PR DESCRIPTION
**Pull request by vincentvanhees**
This PR:
- Adds parameters `dec_reports` and `dec_config` to complement `sep_reports` and `sep_config` that already existed. I have tested this on Dutch configured laptop and it now works smoothly, fixes #1048.
- While doing this I also tidied up the way `params_output` is passed on to some of the functions.
- Adds `Sys.setlocale("LC_TIME", "C")` in a more systematic way. For part 2 and 5 it turned out to be necessary to also set this inside the functions that are run in parallel, because somehow R does not pass on this setting to the parallel process, fixes #1047.
- I observed that parameter `week_weekend_aggregate.part5` was not used, I have now made sure it is actually used.
- Parameters `sep_reports` and `sep_config`, were not checked for being a character by check_params, now also fixed.

Checklist before merging:

- [ ] Existing tests still work (check by running the test suite, e.g. from RStudio).
- [ ] Added tests (if you added functionality) or fixed existing test (if you fixed a bug).
- [x] Updated or expanded the documentation.
- [x] Updated release notes in `inst/NEWS.Rd` with a user-readable summary. Please, include references to relevant issues or PR discussions.
- [ ] Added your name to the contributors lists in the `DESCRIPTION` file, if you think you made a significant contribution.

### CODE REVIEW

**Comment by vincentvanhees** [comment]
@jhmigueles I tested this on a Dutch laptop and have now asked the group in Paris to test it on their computer. However, maybe you want to inform the group you developed the `sep_reports` argument for to start using these new `dec_*` parameters and get them to test it to?

@l-k- Previously I was setting `LC_time` to `C` in every specific instance of GGIR code where language is important, with this PR I am attempting to set it more broadly such that we do not always have to worry about language issues. May I ask you to have a look at how I now enforce `LC_time`. More specifically, I am not sure how to pass on system settings via foreach, so I decided to re-enforce it also inside the subfunctions `main_part2` and `main_part5` inside `g.part2` and `g.part5`. Does this make sense to you or can you see a more elegant way to do this?

File: R/g.getstarttime.R [hunk 1]
```diff
@@ -127,7 +127,6 @@ g.getstarttime = function(datafile, P, header, mon, dformat, desiredtz, configtz
         }
       }
       expectedformat = paste0('%',splitformat[1],sepa,'%',splitformat[2],sepa,'%',splitformat[3],' %H:%M:%S')
-      Sys.setlocale("LC_TIME", "C")  # set language to English because that is what we use elsewhere in GGIR
```
[the remark this answers is not in the document]
Comment (vincentvanhees) [reply R/g.getstarttime.R]: Thank you for spotting this. I have now addressed in commit [3b6ba44](https://github.com/wadpac/GGIR/pull/1049/commits/3b6ba4430816d0e41acb8c44e45c57e8d62d48a4).

> I think this edit got overwritten when you merged my big part 1 PR. 

The line is also in your branch: https://github.com/wadpac/GGIR/blob/part1-simplification-v2/R/g.getstarttime.R So, I am not sure where this went wrong. I had my branch (issue1047_force...) before I merged yours with master, and then rebased my branch where `g.getstarttime` was the only function that required me to resolve a conflict, which I think went well.

File: R/convertEpochData.R [hunk 2]
```diff
@@ -4,7 +4,6 @@ convertEpochData = function(datadir = c(), metadatadir = c(),
   # to a format that allows GGIR to process them as if they were part 1 output.
   
   # Capture current local settings
-  Syslocale = Sys.getenv("LC_TIME")
   Sys.setlocale("LC_TIME", "C") # set language to English
```
[the remark this answers is not in the document]
Comment (vincentvanhees) [reply R/convertEpochData.R]: Thanks, now addressed in commit [3b6ba44](https://github.com/wadpac/GGIR/pull/1049/commits/3b6ba4430816d0e41acb8c44e45c57e8d62d48a4)

File: R/GGIR.R [hunk 3]
```diff
@@ -453,4 +458,6 @@ GGIR = function(mode = 1:5, datadir = c(), outputdir = c(),
               verbose = TRUE)
     }
   }
+  # Reset langagues setting back to what they were4
+  Sys.setlocale("LC_TIME", LC_TIME_backup)
```
[the remark this answers is not in the document]
Comment (vincentvanhees) [reply R/GGIR.R]: Thanks, now addressed in commit [3b6ba44](https://github.com/wadpac/GGIR/pull/1049/commits/3b6ba4430816d0e41acb8c44e45c57e8d62d48a4)

File: R/g.part2.R [hunk 4]
```diff
@@ -83,7 +83,8 @@ g.part2 = function(datadir = c(), metadatadir = c(), f0 = c(), f1 = c(),
                         params_phyact = c(), params_output = c(), params_general = c(),
                         path, ms2.out, foldername, fullfilenames, folderstructure, referencefnames,
                         daySUMMARY, pdffilenumb, pdfpagecount, csvfolder, cnt78, verbose) {
-    
+    # Set language to English while using GGIR
+    Sys.setlocale("LC_TIME", "C")
```
[the remark this answers is not in the document]
Comment (vincentvanhees) [reply R/g.part2.R]: Thanks, that makes sense. Now added in commit [3b6ba44](https://github.com/wadpac/GGIR/pull/1049/commits/3b6ba4430816d0e41acb8c44e45c57e8d62d48a4)

File: R/g.getstarttime.R (hunk 1 above)
[the remark this answers is not in the document]
Comment (l-k-) [reply R/g.getstarttime.R]: Yeah, git normally does a surprisingly great job merging and rebasing, but (thankfully, super rarely) I've seen things like this happen.

File: R/g.part3.R [hunk 5]
```diff
@@ -147,6 +147,7 @@ g.part3 = function(metadatadir = c(), f0, f1, myfun = c(),
                                 varlist = c(unclass(lsf.str(envir = asNamespace("GGIR"), all = T)),
                                             "MONITOR", "FORMAT"),
                                 envir = as.environment(asNamespace("GGIR")))
+        parallel::clusterEvalQ(cl, Sys.setlocale("LC_TIME", "C"))
```
Comment (l-k-) [inline R/g.part3.R:150]: nice! Maybe part 6 too? Or can this definitely never apply to part 6?

Comment (vincentvanhees) [reply R/g.part3.R:150]: Yes, but I have now committed that to a different open PR which comes with many other updates to part6, in that way I the merge will go easier [16b97f0](https://github.com/wadpac/GGIR/pull/1028/commits/16b97f05ca0740851ffd3583230f161cacd9da42)

**Pull request merged**